Readme.md

```diff
@@ -65,6 +65,7 @@ nix-env -if https://github.com/DavHau/mach-nix/tarball/2.3.0 -A mach-nix
 ### Build a virtualenv-style python environment from a requirements.txt
 ```bash
 mach-nix env ./env -r requirements.txt
+# mach-nix env -r ./requirements.txt -p 3.7 --nixpkgs '{ "rev": "c59ea8b8a0e7f927e7291c14ea6cd1bd3a16ff38", "sha256": "1ak7jqx94fjhc68xh1lh35kh3w3ndbadprrb762qgvcfb8351x8v", }' "./env"
```

Comment:
what was the way I did it first
but the handling would be easier with a json 
- if I myself would query the rev like 

```
nixpkgs_version=$(nix-shell -p nix-prefetch-git --run  "nix-prefetch-git --url https://github.com/nixos/nixpkgs-channels.git --rev refs/heads/nixos-unstable  --no-deepClone  --quiet " |  tr -d "\n" )

mach-nix gen -p 3.7 --nixpkgs "${nixpkgs_version}" -r ./requirements.txt # or as one liner
```

---

my aim was to do as little changes to your code/style as possible

if you only get the rev 
- would resolve in an additional dependency - to resolve the sha out of the rev (which is compulsory, right?)
-- [if it would be a nix tool (because you would like to do it in 100% nix) this would break the option of pip install, right? ]

---

For me json gives possibilities for future changes because it is flexible/"schema-less" 
-> json5 would be even better

Reply:
Mach-nix requires nix, even when installed via pip. If nix is not installed, mach-nix will fail and instruct the user to first install nix. Therefore nix can be assumed available.

The command line mach-nix tool is mainly intended for users which are not familiar to nix. Therefore I'd like to keep it as simple as possible. Requiring knowledge about an additional nix tool would not be optimal.

Including nix-prefetch-git as a dependency adds a closure size of around 400 MB since it depends on git. But It can be done in a way, so nix-prefetch-git is only aquired the moment the '--nixpkgs' feature is used. But probably it's not the best idea.

An alternative would be to just use:
1. `git ls-remote https://github.com/nixos/nixpkgs-channels/ {branch}` to receive the current commit
2. `nix-prefetch-url --unpack https://github.com/nixos/nixpkgs-channels/{commit}` to receive the current sha256

nix-prefetch-url is available on each standard nix distribution. Therefore no extra dependencies are required.

Concerning the future solution. Flakes should probably be used for managing and updating the nixpkgs and pypi-deps-db dependency. But currently flakes requires an unstable version of nix and which I wouldn't want to force on all mach-nix users.

Since it was my mistake resulting in delaying the PR so much, I'd like to come towards you. If you like, I'm OK to merge it like it is. I can improve it myself in a future commit.

Reply:
I would argue that a beginner will not need/use the feature to change nixpkgs version

---

you are right, that it can be fetched easy via github api 

```
owner="DavHau"
repo="mach-nix"
httpsLink="https://api.github.com"
curl -i -H "Accept: application/json" -H "Content-Type: application/json" -X GET "${httpsLink}"/repos/"${owner}"/"${repo}"/commits > test_curl_githubApi.txt
```

should be a short python script (when you don't want to implement it in nix [that could be more work.-] )

---

it is you product - if you don't like it don't merge it

Reply:
Doing this in python sounds like a good idea to me. There wouldn't be any benefit of doing that in nix. Aside, fetching stuff without knowing the outcome isn't even possible in nix